src/lib.ts

```diff
@@ -0,0 +1,320 @@
+import sharp from 'sharp';
+import smartcrop from 'smartcrop-sharp';
+import imageminPngquant from 'imagemin-pngquant';
+import { S3Client, S3ClientConfig, GetObjectCommand, GetObjectCommandOutput } from '@aws-sdk/client-s3';
+
+const authenticatedRequest = !!process.env.S3_AUTHENTICATED_REQUEST
+	? process.env.S3_AUTHENTICATED_REQUEST.toLowerCase() == 'true'
+	: false;
+
+export interface Args {
+	webp?: string | boolean;
+	background?: string;
+	lb?: string;
+	zoom?: string;
+	crop?: string | string[];
+	fit?: string;
+	gravity?: string;
+	crop_strategy?: string;
+	resize?: string | number[];
+	quality?: string | number;
+	h?: string;
+	w?: string;
+	key: string;
+	'X-Amz-Algorithm'?: string;
+	'X-Amz-Content-Sha256'?: string;
+	'X-Amz-Credential'?: string;
+	'X-Amz-SignedHeaders'?: string;
+	'X-Amz-Expires'?: string;
+	'X-Amz-Signature'?: string;
+	'X-Amz-Date'?: string;
+	'X-Amz-Security-Token'?: string;
+}
+
+function getDimArray(dims: string | number[], zoom: number = 1): (number | null)[] {
+	var dimArr = typeof dims === 'string' ? dims.split(',') : dims;
+	return dimArr.map(function (v) {
+		return Math.round(Number(v) * zoom) || null;
+	});
+}
+
+function clamp(val: number | string, min: number, max: number): number {
+	return Math.min(Math.max(Number(val), min), max);
+}
+
+export async function getS3File(
+	config: S3ClientConfig & { bucket: string },
+	key: string,
+	args: Args
+): Promise<GetObjectCommandOutput> {
+	const s3 = new S3Client({
+		...config,
+		signer: {
+			sign: async (request) => {
+				const isPresigned = !!args['X-Amz-Algorithm'];
+				if (!isPresigned) {
```

Comment:
Unneeded boolean cast and var.

```suggestion
				if (!args['X-Amz-Algorithm']) {
```